### PR TITLE
[IMP] web_menu_navbar_needaction: reflow, new features

### PR DESCRIPTION
**Pull request by hbrunn**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
:+1:

**Comment by hbrunn** [comment]
added another commit to this branch which adds a few features

File: web_menu_navbar_needaction/static/src/js/web_menu_navbar_needaction.js [hunk 1]
```diff
@@ -50,29 +50,57 @@ openerp.web_menu_navbar_needaction = function(instance)
...
+                            ev.preventDefault();
+                            return parent.menu_dm.add(
+                                self.rpc('/web/action/load', {
+                                action_id: needaction_data.action_id,
+                            }))
+                            .then(function(action)
+                            {
+                                return parent.action_mutex.exec(function()
+                                {
+                                    action.domain = needaction_data.action_domain;
```
Comment (hbrunn) [inline web_menu_navbar_needaction/static/src/js/web_menu_navbar_needaction.js:95]: we also need to filter out `search_default_` context keys here in order not to interfere with the search domain, and turn off custom filters: https://github.com/OCA/OCB/blob/8.0/addons/web/static/src/js/search.js#L374

Comment (hbrunn) [inline web_menu_navbar_needaction/static/src/js/web_menu_navbar_needaction.js:95]: done in 33db0df

**Comment by hbrunn** [comment]
@OCA/web-maintainers the last commit adds the seriously cool feature to let you assign needaction counters to any menu item, not just for models that support it

**Comment by OSguard** [comment]
@codingforfun can you review this

**Comment by gurneyalex** [comment]
:+1:

**Comment by hbrunn** [comment]
@gurneyalex @pedrobaeza ruthlessly rebased, green, and you two approved quite a while ago :-)

**Comment by pedrobaeza** [comment]
Do you see interesting to preserve commit history or can I squash them?

**Comment by hbrunn** [comment]
I invested some love for sorting the commits, but given none of this lives in 9 and probably never will, it's not really necessary to preserve the logical units I think. Go ahead as you prefer

**Comment by pedrobaeza** [comment]
I prefer to squash because this way, when migrating to later version, we have a cleaner commit history and it's easier to identify changes.

**Pull request merged**